packages/driver/cypress/support/utils.js

```diff
@@ -94,6 +108,10 @@ const getAllFn = (...aliases) => {
   )
 }
 
+const shouldWithTimeout = (cb, timeout = 250) => {
```

Comment:
Just curious, why is this custom command needed?

Reply:
cy.should uses the timeout from the previous command. You can't specify it for the cy.should itself. Since cy.origin doesn't have a timeout, it defaults to 0 for cy.should, which means it doesn't retry at all. This allows specifying a timeout (or using the default 250ms) for cy.should to use.

Might be good to add a code comment here explaining that.

Reply:
Just a convenience thing that @chrisbreiding came up with [here](https://github.com/cypress-io/cypress/pull/20949#discussion_r850457142). He explains it a lot better in the thread than I can.

Reply:
There's no need for a timeout option for cy.origin. It wouldn't have any role in how cy.origin operates. 

This is more of an issue with cy.should. Normally, cy.should will retry the previous command if it's something like a dom query, so it makes sense for it to use that command's timeout. In the cases where we're using this custom command, we're not retrying cy.origin, we retrying until all the relevant logs have been gathered. I'm not sure it's applicable to most users' use-cases. Perhaps cy.should should accept a timeout argument so something like this isn't necessary.

Reply:
I was told cy.origin will yield anything that is serializable, so I would expect the .should to work when chained off of the command....like:
```
cy.origin('foo.com', () => {
    // do stuff
    return cy.url()
}).should('eq', 'www.foo.com')
```

It looks like we have [tests in place like this](https://github.com/cypress-io/cypress/blob/feature-multidomain/packages/driver/cypress/integration/e2e/multi-domain/multi_domain_yield_spec.ts#L17), so why the timeout is 0 in this instance?

Reply:
I was wrong about the timeout having anything to do with it. cy.should does not retry. Instead, any given command controls whether or not it retries. It will retry its own code and retry any upcoming assertions (which often includes re-running cy.should). Re-trying cy.get, for example, will re-query the dom for an element and then retry any assertions on that element. It has to be manually implemented per command in order to specify what gets retried and [looks like this in the case of cy.get](https://github.com/cypress-io/cypress/blob/79267ddc0fdc5e82fcd2fba2e08ecd1e7a5c362e/packages/driver/src/cy/commands/querying/querying.ts#L179-L199).

Currently, cy.origin does not retry, which is what causes the cy.should not to retry. cy.wrap _does_ retry, so we can get the subsequent cy.should to retry using it.

We honestly haven't talked about retry-ability for cy.origin, but I think it probably wouldn't work to retry the entire callback because, at that point, the page may have navigated and the AUT is no longer on that origin. It would be a better pattern for users to make any necessary assertions within the cy.origin callback. Our tests aren't really good examples of user tests since they're testing something specific about Cypress itself. For the cases where we're using shouldWithTimeout, retrying cy.origin wouldn't be useful, since it's the logs we're waiting for, not a specific result yielded from cy.origin.

For the test case you quoted, yes, cy.should works, but it won't retry. It passes because the assertion passes the one and only time it's run.

Reply:
Gotchya that makes more sense on how this is behaving and why we want to wrap like this.

I'd agree, retrying the whole whole cy.origin command might not make sense. This makes me think of an issue I've hit randomly with aliases though. I assume given everything else I've heard, aliases will not be referenced in the origin callback, however if there is something like:

```js
cy.get('button').aslias(my_button)
cy.get('@my_butto').click() // navigates to diff orgin

cy.origin('....', () => do something)

cy.get('@my_button').should('be.visible') // because button does not exist, the commands  are re-queued backwards to re-run the button to try and re-find the element that was alised
```

Will the above work okay?

Reply:
Assuming that the `cy.origin` redirects back to local host, yes that works fine (I tested). If `cy.origin` doesn't redirect back the command will timeout.